src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/SubmitAppealPreparer.java

```diff
@@ -0,0 +1,87 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.APPEAL_TYPE;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.PAY_FOR_THE_APPEAL_OPTION;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.PA_APPEAL_TYPE_PAYMENT_OPTION;
+
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AppealType;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.JourneyType;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+
+@Component
+public class SubmitAppealPreparer implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final boolean isfeePaymentEnabled;
+
+    public SubmitAppealPreparer(
+        @Value("${featureFlag.isfeePaymentEnabled}") boolean isfeePaymentEnabled
+    ) {
+        this.isfeePaymentEnabled = isfeePaymentEnabled;
+    }
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        boolean isRepJourney = callback.getCaseDetails().getCaseData()
+            .read(AsylumCaseFieldDefinition.JOURNEY_TYPE, JourneyType.class)
+            .map(journeyType -> journeyType == JourneyType.REP)
+            .orElse(true);
+
+        return (callbackStage == PreSubmitCallbackStage.ABOUT_TO_START)
+               && Event.SUBMIT_APPEAL == callback.getEvent()
+               && isfeePaymentEnabled
+               && isRepJourney;
```

Comment:
Hi @alexbaeza just FYI,
we are adding fees and payments feature check so Rep journey will go through PAY_AND_SUBMIT event
I guess AIP stays as it is within this boolean flag check